OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -330,10 +335,21 @@ protected static List<int> CalculateFrameIndices(int start, int? length, int str
 
 		protected virtual IEnumerable<ReservationInfo> ParseFilenames(ModData modData, string tileset, int[] frames, MiniYaml data, MiniYaml defaults)
 		{
+			var filenamePatternNode = data.NodeWithKeyOrDefault(FilenamePattern.Key) ?? defaults.NodeWithKeyOrDefault(FilenamePattern.Key);
+			if (!string.IsNullOrEmpty(filenamePatternNode?.Value.Value))
+			{
+				var patternStart = LoadField("Start", 0, filenamePatternNode.Value);
+				var patternCount = LoadField("Count", 1, filenamePatternNode.Value);
+
+				return Enumerable.Range(patternStart, patternCount).Select(i =>
+					new ReservationInfo(filenamePatternNode.Value.Value.FormatInvariant(i),
+						FirstFrame, FirstFrame, filenamePatternNode.Location));
+			}
+
 			var filename = LoadField(Filename, data, defaults, out var location);
 
 			var loadFrames = CalculateFrameIndices(start, length, stride ?? length ?? 0, facings, frames, transpose, reverseFacings, shadowStart);
-			yield return new ReservationInfo(filename, loadFrames, frames, location);
+			return new[] { new ReservationInfo(filename, loadFrames, frames, location) };
```

Comment:
The old code made the function a generator, which wasn't compatible with the new code added above (which would otherwise have to explicitly loop over the results and `yield` them). The subclasses already returned explicit enumerables so this also improves consistency.

Reply:
Ah, right, the code above. Derp. Still not ideal but ok.